frontend/lib/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -269,21 +367,164 @@ export class ArrowVegaLiteChart extends PureComponent<
       )
     ) {
       if (prevNumRows < numRows) {
+        // Insert the new rows.
         this.vegaView.insert(name, getDataArray(data, prevNumRows))
       }
     } else {
       // Clean the dataset and insert from scratch.
-      const cs = vega
-        .changeset()
-        .remove(vega.truthy)
-        .insert(getDataArray(data))
-      this.vegaView.change(name, cs)
+      this.vegaView.data(name, getDataArray(data))
       logMessage(
         `Had to clear the ${name} dataset before inserting data through Vega view.`
       )
     }
   }
 
+  /**
+   * Configure the selections for this chart if the chart has selections enabled.
+   */
+  private maybeConfigureSelections = (): void => {
+    if (this.vegaView === undefined) {
+      // This check is mainly to make the type checker happy.
+      // this.vegaView is guaranteed to be defined here.
+      return
+    }
+
+    const { widgetMgr, element } = this.props
+
+    if (!element?.id || element.selectionMode.length === 0) {
+      // To configure selections, it needs to be activated and
+      // the element ID must be set.
+      return
+    }
+
+    // Try to load the previous state of the chart from the element state.
+    // This is useful to restore the selection state when the component is re-mounted
+    // or when its put into fullscreen mode.
+    const viewState = widgetMgr.getElementState(
+      this.props.element.id,
+      "viewState"
+    )
+    if (notNullOrUndefined(viewState)) {
+      try {
+        this.vegaView = this.vegaView.setState(viewState)
+      } catch (e) {
+        logWarning("Failed to restore view state", e)
+      }
+    }
+
+    // Add listeners for all selection events. Find out more here:
+    // https://vega.github.io/vega/docs/api/view/#view_addSignalListener
+    element.selectionMode.forEach((param, _index) => {
+      this.vegaView?.addSignalListener(
+        param,
+        debounce(DEBOUNCE_TIME_MS, (name: string, value: SignalValue) => {
+          // Store the current chart selection state with the widget manager so that it
+          // can be used for restoring the state when the component unmounted and
+          // created again. This can happen when elements are added before it within
+          // the delta path.
+          const viewState = this.vegaView?.getState({
+            // There are also `signals` data, but I believe its
+            // not relevant for restoring the selection state.
+            data: (_name?: string, _operator?: any) => {
+              // Vega lite stores the selection state in a <param name>_store parameter
+              // under `data` that can be retrieved via the getState method.
+              // https://vega.github.io/vega/docs/api/view/#view_getState
+              const stateNames = element.selectionMode.map(
+                (param, _idx) => `${param}_store`
+              )
+              return _name ? stateNames.includes(_name) : false
```

Comment:
why do we first construct a dict here? can we not iterate through the `selectionMode` like

```typescript
for (const mode of selectionMode) {
  if (_name === `${mode}_store`) return true
  return false
}
```

Reply:
Good point, I simplified it even more to: 
```typescript
return element.selectionMode.some(mode => `${mode}_store` === name)
```